embrace-android-core/src/main/kotlin/io/embrace/android/embracesdk/internal/capture/session/EmbraceSessionProperties.kt

```diff
@@ -82,7 +94,23 @@ internal class EmbraceSessionProperties(
 
     private fun size(): Int = permanentProperties().size + temporary.size
 
-    fun clearTemporary(): Unit = temporary.clear()
+    fun clearTemporary() {
+        synchronized(permanentPropertiesReference) {
+            temporary.clear()
+            addPermPropsToSessionSpan()
```

Comment:
I think we need to add the permanent session properties at the beginning, when the span is started, because the session could be killed before it ends, and we need to send the cached version over. Is there another place we can do this population that doesn't bring back to circular dependency?

Reply:
The `permanentProperties()` function (L29) should take care of properties being added to the very first session of a process. Unless I've misunderstood something?